From 68ed0ca853338c9436d5047777f3f395c0f2c5b2 Mon Sep 17 00:00:00 2001
From: Nolann Chobert <nolann.chobert@nxp.com>
Date: Mon, 28 Jul 2025 16:00:19 +0000
Subject: [PATCH] Fix libnnstreamer_customfilter_passthrough.so path

Custom filter path is set to Yocto build path which will trigger an error on target.
The appropriate path is /usr/lib/nnstreamer/customfilters/libnnstreamer_customfilter_passthrough.so.

Upstream-Status: Inappropriate [i.MX specific]

Signed-off-by: Nolann Chobert <nolann.chobert@nxp.com>
---
 tests/nnstreamer_filter_extensions_common/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/nnstreamer_filter_extensions_common/meson.build b/tests/nnstreamer_filter_extensions_common/meson.build
index ecd1878b..eba2cc83 100644
--- a/tests/nnstreamer_filter_extensions_common/meson.build
+++ b/tests/nnstreamer_filter_extensions_common/meson.build
@@ -9,7 +9,9 @@ tizen_apptest_deps = [
 # Format for adding subplugin into extensions -
 # [name, extension abbreviation, dependencies, model file name/folder path/file path, test name]
 extensions = []
-custom_filter_path = join_paths(meson.build_root(), 'tests', 'nnstreamer_example',
+nnst_prefix = get_option('prefix')
+nnst_libdir = join_paths(nnst_prefix, get_option('libdir'))
+custom_filter_path = join_paths(nnst_libdir, 'nnstreamer', 'customfilters',
     'libnnstreamer_customfilter_passthrough.' + so_ext)
 extensions += [['custom', 'custom', nnstreamer_unittest_deps, custom_filter_path, 'custom']]
 extensions += [['custom', 'custom', nnstreamer_unittest_deps, custom_filter_path, 'custom-set']]
-- 
2.34.1

